src/hotspot/share/runtime/os.cpp

```diff
@@ -882,6 +882,53 @@ void os::abort(bool dump_core) {
 //---------------------------------------------------------------------------
 // Helper functions for fatal error handler
 
+bool os::print_function_and_library_name(outputStream* st,
+                                         address addr,
+                                         char* buf, int buflen,
+                                         bool shorten_paths,
+                                         bool demangle,
+                                         bool strip_arguments) {
+  // If no scratch buffer given, allocate one here on stack.
+  // (used during error handling; its a coin toss, really, if on-stack allocation
+  //  is worse than (raw) C-heap allocation in that case).
+  char* p = buf;
+  if (p == NULL) {
+    p = (char*)::alloca(O_BUFLEN);
+    buflen = O_BUFLEN;
+  }
+  int offset = 0;
+  const bool have_function_name = dll_address_to_function_name(addr, p, buflen,
+                                                               &offset, demangle);
+  if (have_function_name) {
+    if (demangle && strip_arguments) {
+      char* args_start = strchr(p, '(');
+      if (args_start != NULL) {
+        *args_start = '\0';
+      }
+    }
+    st->print("%s+%d", p, offset);
```

Comment:
Minor nit -- maybe we should do this to avoid print "+0" all the time?

```
if (offset == 0) {
  st->print("%s");
} else {
  st->print("%s+%d", p, offset);
}
```

Reply:
You are right, that's better.